File: docs/src/testers/functions/invoke_function.py
# @sniptest filename=invoke_function.py
# @sniptest show=6-18
# ruff: noqa: E402 -- hidden live fixture precedes displayed imports
import os

function_id = os.environ["NOTTE_FUNCTION_ID"]

from notte_sdk import NotteClient

client = NotteClient()

# Get function by ID
function = client.Function(function_id=function_id)

# Run function with parameters
result = function.run(url="https://example.com", search_query="laptop")

print(result.result)  # Access the return value
print(result.status)  # "closed" or "failed"
print(result.session_id)  # Session ID if created

assert result.function_id == function_id

File: docs/src/testers/functions/invoke_function.ts
// @sniptest filename=invoke_function.ts
// @sniptest show=5-17
import assert from 'node:assert/strict';

const functionId = process.env.NOTTE_FUNCTION_ID!;

import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

// Get function by ID
const fn = client.NotteFunction({ function_id: functionId });

// Run function with parameters
const result = await fn.run({ url: 'https://example.com', search_query: 'laptop' });

console.log(result.result); // Access the return value
console.log(result.status); // "closed" or "failed"
console.log(result.session_id); // Session ID if created

assert.equal(result.function_id, functionId);
export { result };
